opentelemetry-otlp/src/exporter/http/mod.rs

```diff
@@ -287,7 +287,7 @@ impl OtlpHttpClient {
     fn build_trace_export_body(
         &self,
         spans: Vec<SpanData>,
-    ) -> opentelemetry_sdk::trace::TraceResult<(Vec<u8>, &'static str)> {
+    ) -> Result<(Vec<u8>, &'static str), String> {
```

Comment:
I don't know how strongly I feel about this, but it smells a bit to me to have `Result<T, String>` and not `Result<T, SomeEnumErrorType` where the thing implements the `Error` trait (presumably with anyhow) , _even though the error type would wrap a single variant_.

This would of course make no difference in the behaviour in this instance, but it would look more consistent, and mean we continue to only return `Error`-implementing things in `Result`.

Reply:
This is just internal helper functions, so okay with this. But for public API, lets ensure Result<T, SomeError> will be returned always.